src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -173,6 +173,12 @@ public virtual string CustomMethod
         #region Proxy
 
         /// <summary>
+        /// gets or sets the NoProxy property
+        /// </summary>
+        [Parameter]
+        public virtual SwitchParameter NoProxy { get; set; }
```

Comment:
It seems that it is better not to do different parameter sets because we will not be able to do:
```powershell
function Foo($NoProxy) {
   Invoke-WebRequest -NoProxy:$NoProxy -Proxy "http://proxy.server.com"
}
```